### PR TITLE
Add payload event

### PR DESCRIPTION
**Pull request by Sly-Little-Fox**
### Changes

-   `adds`: A `payload` event, which forwards any gateway messages.

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

Recreated #74 because of a GitHub :bug:.

### CODE REVIEW

**Review by Arthurdw — APPROVED** [review]
Ready for merge I think

**Comment by Arthurdw** [comment]
Thanks as always for the contributions ♥

File: pincer/client.py [hunk 1]
```diff
@@ -162,9 +162,10 @@ def __init__(
             token,
             handlers={
                 # Use this event handler for opcode 0.
+                -1: self.payload_event_handler,
                 0: self.event_handler
             },
-            intents=intents or Intents.NONE
+            intents=intents or Intents.NONE,
```
Comment (Sigmanificient) [inline pincer/client.py:168]: The useless comma has returned!

File: pincer/client.py [hunk 2]
```diff
@@ -162,9 +162,10 @@ def __init__(
             token,
             handlers={
                 # Use this event handler for opcode 0.
+                -1: self.payload_event_handler,
```
Comment (Sigmanificient) [inline pincer/client.py:165]: This is so cursed, but i guess it works

File: pincer/client.py [hunk 3]
```diff
@@ -116,7 +116,7 @@ def decorator(func: Coro):
         async def wrapper(cls, payload: GatewayDispatch):
             _log.debug("`%s` middleware has been invoked", call)
 
-            print(func, should_pass_cls(func))
+            # print(func, should_pass_cls(func))
```
Comment (Sigmanificient) [inline pincer/client.py:119]: may be removed

File: pincer/client.py (hunk 1 above)
Comment (Sigmanificient) [inline pincer/client.py:168]: The useless comma, .... as returned!

File: pincer/client.py (hunk 2 above)
Comment (Arthurdw) [reply pincer/client.py:165]: -1 represents infinity, so it will include everything

File: pincer/client.py (hunk 3 above)
Comment (Arthurdw) [reply pincer/client.py:119]: Resolved.

File: pincer/client.py (hunk 1 above)
Comment (Arthurdw) [reply pincer/client.py:168]: Resolved.

**Pull request merged**